docs/grfn_spec.md

```diff
@@ -11,96 +20,189 @@ associated extracted comments, links to natural language text (the
+- \'FOR NOW\': Tags approach being currently taken, eiher in response to FUTURE or CHOICE.
+
+### From source code to dynamic system representation
+
+The goal of GrFN is to provide the end-point target for a translation from the semantics of program (computation) specification (as asserted in source code) to the semantics of a (discretized) dynamic system model.
+
+A key assumption is that the program source code we are analyzing is intended to model aspects of some target physical domain, and that this target physical domain is a dynamical system that evolves over time.
+
+The system is decomposed into a set of individual states (represented as random variables), where the values of the states at any given time are a function of the values of zero or more other states at the current and/or previous time point(s). Because we are considering the evolution of the system over time, in general every variable has an index. The functional relationships may be instantaneous (based on the variables indexed at the same point in time) or a function of states of variables at different time indices.
+
+
+Identifiers: grounding, scopes, namespaces and gensyms
+------------------------------------------------------
+
+An identifier is a symbol used to uniquely identify a program element in code, where a *program element* is a 
+- variable (or constant)
+- function
+- type (class)
+
+More than one identifier can be used to denote the same program element, but an identifier can only be associated with one program element at a time.
+
+### Grounding
+
+Identifiers play a key role in connecting the model as implemented in source code to the target domain that it models. *Grounding* is the task of inferring what aspect of target domain a program element may correspond to, and identifiers, by their (base) name(s), their declaration and use (where they occur in code -- `scope` and `namespace`), and the doc and comment strings that occur around them, provide clues to what they are intended to refer to. For this reason, we need to associate with identifiers several pieces of information. This information will be collected during program analysis and associated with the identifier declaration:
+
+- "aliases": It is possible for multiple identifiers to be used to denote the same program element. How this is done differs across languages, according to scoping rules and assignment. Program analysis modules for each language (e.g., the Fortran `for2py` analyzer) will determine this. One general way to assign more than one identifier to the same program element is through a *simple equality assignment*, e.g.: `y = x` means that a new identifier, `y`, denotes the same program element that `x` does. A simple equality assignment just involve one identifier being equated with another, no other operations are applied; if other operations are applied (e.g., `y = x + 1`), then this is a *new* identifier as it does not represent the original value of `x` but a modification of it.
```

Comment:
"aliases": ... "Program analysis modules for each language (e.g., the Fortran `for2py` analyzer) will determine this."  Note that once we have pointers (and DSSAT seems to have them) it can become impossible in general to determine all aliases statically.  Suggest making this a FUTURE issue.

Reply:
Good point. Added this note:
> FUTURE: Note that once we consider pointers (e.g., DSSAT has some), it can become impossible in general to determine all aliases *statically*.